tests/test_response_processor.py

```diff
@@ -255,6 +255,36 @@ def test_send_receipt(self):
         with self.assertRaises(QuarantinableError):
             self.rp.send_receipt(invalid_json)
 
+    def test_send_rm_receipt(self):
+        self.rp.decrypt_survey = MagicMock(return_value=valid_rm_json)
```

Comment:
The expected test case should be :

    1.QuarantinableError in case of KeyError on receipt_json
    2.QuarantinableError in case of no decrypted_json.get("survey_id")
    3.return None in case of decrypted_json.get("survey_id") == "census"
    4.PublishMessageError ( RetryableError ) in case of self.rrm_publisher.publish
    5. Positive case for all of them .

Reply:
The underlying functionality is already tested in this module, apart from number 3, which isn't explicitly covered but was executed as part of a test for specific log lines (I've now added this).